src/test/java/uk/gov/companieshouse/items/orders/api/controller/CertificateItemsControllerUnitTest.java

```diff
@@ -31,48 +36,64 @@
-        newCertificateItemDTO.setCompanyNumber("1234");
+        final CertificateItemDTO newItem = new CertificateItemDTO();
+        newItem.setCompanyNumber("1234");
         final CertificateItemOptions options = new CertificateItemOptions();
         options.setCertInc(true);
         options.setCertShar(true);
-        newCertificateItemDTO.setItemOptions(options);
-        newCertificateItemDTO.setQuantity(5);
+        newItem.setItemOptions(options);
+        newItem.setQuantity(5);
 
-        final CertificateItemDTO createdCertificateItemDTO = new CertificateItemDTO();
-        createdCertificateItemDTO.setId("CHS1");
-        createdCertificateItemDTO.setCompanyNumber(newCertificateItemDTO.getCompanyNumber());
-        createdCertificateItemDTO.setKind("certificate");
-        createdCertificateItemDTO.setDescriptionIdentifier("certificate");
+        final CertificateItemDTO expectedItem = new CertificateItemDTO();
+        expectedItem.setId(EXPECTED_ITEM_ID);
+        expectedItem.setCompanyNumber(newItem.getCompanyNumber());
+        expectedItem.setKind("certificate");
+        expectedItem.setDescriptionIdentifier("certificate");
         final ItemCosts costs = new ItemCosts();
         costs.setDiscountApplied("1");
         costs.setIndividualItemCost("2");
         costs.setPostageCost("3");
         costs.setTotalCost("4");
-        createdCertificateItemDTO.setItemCosts(costs);
-        createdCertificateItemDTO.setItemOptions(options);
-        createdCertificateItemDTO.setPostalDelivery(true);
-        createdCertificateItemDTO.setQuantity(5);
+        expectedItem.setItemCosts(costs);
+        expectedItem.setItemOptions(options);
+        expectedItem.setPostalDelivery(true);
+        expectedItem.setQuantity(5);
 
         // When and Then
-        mockMvc.perform(post("/certificates")
+        final ResultActions outcome = mockMvc.perform(post("/certificates")
                 .contentType(MediaType.APPLICATION_JSON)
-                .content(objectMapper.writeValueAsString(newCertificateItemDTO)))
+                .content(objectMapper.writeValueAsString(newItem)))
                 .andExpect(status().isCreated())
-                .andExpect(content().json(objectMapper.writeValueAsString(createdCertificateItemDTO)))
+                .andExpect(content().json(objectMapper.writeValueAsString(expectedItem)))
                 .andExpect(jsonPath("$.item_options.cert_inc", is(true)))
                 .andExpect(jsonPath("$.item_options.cert_shar", is(true)))
                 .andExpect(jsonPath("$.item_options.cert_dissliq", is(false)))
-                .andExpect(jsonPath("$.postal_delivery", is(true))).andDo(MockMvcResultHandlers.print());
+                .andExpect(jsonPath("$.postal_delivery", is(true)))
+                .andDo(MockMvcResultHandlers.print());
 
         // Then
-        // TODO PCI-324 Verify can retrieve equivalent entity
+        assertItemSavedCorrectly(EXPECTED_ITEM_ID);
+    }
```

Comment:
Unit tests are missing for failure scenarios, e.g. validation failure

Reply:
Good point. Will add.